File: app/CerealExercise/CerealExercise/Theme/AppTheme.swift
import SwiftUI

enum AppTheme: String, CaseIterable, Identifiable, Sendable {
    case peach          // デフォルト (ピーチ / コーラル) — 女性向け
    case sky            // スカイ (ブルー / ネイビー) — 男性向け
    case midnight       // ミッドナイト (ダーク) — 暗め
    case sunshine       // サンシャイン (イエロー / オレンジ) — 黄色系
    case forest         // フォレスト (グリーン / アース) — 自然系

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .peach: "ピーチ"
        case .sky: "スカイ"
        case .midnight: "ミッドナイト"
        case .sunshine: "サンシャイン"
        case .forest: "フォレスト"
        }
    }

    var hint: String {
        switch self {
        case .peach: "やわらかピーチ&コーラル"
        case .sky: "クールなブルー系"
        case .midnight: "目に優しい暗め配色"
        case .sunshine: "明るく元気な黄色系"
        case .forest: "落ち着きのある自然系"
        }
    }

    // MARK: - Color tokens

    var background: Color {
        switch self {
        case .peach: Color(red: 1.00, green: 0.97, blue: 0.93)
        case .sky: Color(red: 0.94, green: 0.97, blue: 1.00)
        case .midnight: Color(red: 0.10, green: 0.11, blue: 0.14)
        case .sunshine: Color(red: 1.00, green: 0.98, blue: 0.90)
        case .forest: Color(red: 0.95, green: 0.97, blue: 0.93)
        }
    }

    var surface: Color {
        switch self {
        case .peach: Color(red: 1.00, green: 0.99, blue: 0.96)
        case .sky: Color(red: 1.00, green: 1.00, blue: 1.00)
        case .midnight: Color(red: 0.17, green: 0.18, blue: 0.22)
        case .sunshine: Color(red: 1.00, green: 1.00, blue: 0.96)
        case .forest: Color(red: 1.00, green: 1.00, blue: 0.98)
        }
    }

    var primary: Color {
        switch self {
        case .peach: Color(red: 1.00, green: 0.62, blue: 0.55)
        case .sky: Color(red: 0.36, green: 0.62, blue: 0.92)
        case .midnight: Color(red: 0.65, green: 0.78, blue: 1.00)
        case .sunshine: Color(red: 1.00, green: 0.78, blue: 0.30)
        case .forest: Color(red: 0.42, green: 0.66, blue: 0.52)
        }
    }

    var primaryDeep: Color {
        switch self {
        case .peach: Color(red: 0.84, green: 0.33, blue: 0.30)
        case .sky: Color(red: 0.18, green: 0.40, blue: 0.78)
        case .midnight: Color(red: 0.45, green: 0.62, blue: 0.95)
        case .sunshine: Color(red: 0.85, green: 0.58, blue: 0.12)
        case .forest: Color(red: 0.22, green: 0.50, blue: 0.32)
        }
    }

    var secondary: Color {
        switch self {
        case .peach: Color(red: 0.96, green: 0.85, blue: 0.74)
        case .sky: Color(red: 0.78, green: 0.88, blue: 0.98)
        case .midnight: Color(red: 0.30, green: 0.34, blue: 0.42)
        case .sunshine: Color(red: 0.98, green: 0.90, blue: 0.70)
        case .forest: Color(red: 0.82, green: 0.92, blue: 0.84)
        }
    }

    var textPrimary: Color {
        switch self {
        case .peach: Color(red: 0.30, green: 0.25, blue: 0.20)
        case .sky: Color(red: 0.16, green: 0.20, blue: 0.30)
        case .midnight: Color(red: 0.95, green: 0.95, blue: 0.97)
        case .sunshine: Color(red: 0.35, green: 0.28, blue: 0.12)
        case .forest: Color(red: 0.18, green: 0.26, blue: 0.20)
        }
    }

    var textSecondary: Color {
        switch self {
        case .peach: Color(red: 0.54, green: 0.47, blue: 0.39)
        case .sky: Color(red: 0.40, green: 0.46, blue: 0.55)
        case .midnight: Color(red: 0.70, green: 0.72, blue: 0.78)
        case .sunshine: Color(red: 0.55, green: 0.48, blue: 0.32)
        case .forest: Color(red: 0.42, green: 0.50, blue: 0.42)
        }
    }

    var success: Color {
        switch self {
        case .peach: Color(red: 0.55, green: 0.78, blue: 0.55)
        case .sky: Color(red: 0.36, green: 0.78, blue: 0.66)
        case .midnight: Color(red: 0.40, green: 0.85, blue: 0.65)
        case .sunshine: Color(red: 0.60, green: 0.80, blue: 0.40)
        case .forest: Color(red: 0.45, green: 0.74, blue: 0.48)
        }
    }

    var restDay: Color {
        switch self {
        case .peach: Color(red: 0.70, green: 0.80, blue: 0.95)
        case .sky: Color(red: 0.60, green: 0.78, blue: 0.95)
        case .midnight: Color(red: 0.45, green: 0.55, blue: 0.78)
        case .sunshine: Color(red: 0.78, green: 0.84, blue: 0.95)
        case .forest: Color(red: 0.70, green: 0.85, blue: 0.90)
        }
    }

    var missed: Color {
        switch self {
        case .peach: Color(red: 0.86, green: 0.47, blue: 0.45)
        case .sky: Color(red: 0.85, green: 0.55, blue: 0.55)
        case .midnight: Color(red: 0.95, green: 0.45, blue: 0.55)
        case .sunshine: Color(red: 0.88, green: 0.50, blue: 0.30)
        case .forest: Color(red: 0.78, green: 0.50, blue: 0.45)
        }
    }

    var historyAccent: Color {
        switch self {
        case .peach: Color(red: 0.62, green: 0.49, blue: 0.86)
        case .sky: Color(red: 0.45, green: 0.58, blue: 0.85)
        case .midnight: Color(red: 0.75, green: 0.60, blue: 0.92)
        case .sunshine: Color(red: 0.85, green: 0.55, blue: 0.45)
        case .forest: Color(red: 0.55, green: 0.65, blue: 0.45)
        }
    }

    var settingsAccent: Color {
        switch self {
        case .peach: Color(red: 0.36, green: 0.62, blue: 0.72)
        case .sky: Color(red: 0.30, green: 0.55, blue: 0.78)
        case .midnight: Color(red: 0.55, green: 0.70, blue: 0.85)
        case .sunshine: Color(red: 0.65, green: 0.55, blue: 0.40)
        case .forest: Color(red: 0.45, green: 0.62, blue: 0.55)
        }
    }

    var chipBackground: Color {
        switch self {
        case .peach: Color(red: 1.00, green: 0.91, blue: 0.86)
        case .sky: Color(red: 0.88, green: 0.94, blue: 1.00)
        case .midnight: Color(red: 0.22, green: 0.25, blue: 0.32)
        case .sunshine: Color(red: 1.00, green: 0.94, blue: 0.78)
        case .forest: Color(red: 0.92, green: 0.96, blue: 0.88)
        }
    }

    func categoryColor(for category: WorkoutCategory) -> Color {
        switch category {
        case .cardio: Color(red: 0.36, green: 0.62, blue: 0.72)
        case .strength: primaryDeep
        case .yoga: Color(red: 0.58, green: 0.55, blue: 0.82)
        case .stretch: Color(red: 0.42, green: 0.66, blue: 0.52)
        case .fasciaRelease: Color(red: 0.85, green: 0.55, blue: 0.40)
        case .other: historyAccent
        }
    }

    /// Preferred system color scheme for this theme — lets SwiftUI switch
    /// default control tints to a sensible default.
    var preferredColorScheme: ColorScheme? {
        switch self {
        case .midnight: return .dark
        default: return .light
        }
    }
}

File: app-android/app/src/main/java/com/goexercise/app/ui/theme/AppTheme.kt
package com.goexercise.app.ui.theme

import androidx.compose.ui.graphics.Color
import com.goexercise.app.domain.WorkoutCategory

/**
 * アプリのカラーテーマ。iOS `AppTheme.swift` の 1:1 移植(5テーマ × 全カラートークン)。
 * 色値は iOS の `Color(red:green:blue:)`(0..1)をそのまま Float で転記している。
 * 仕様一致が要件なので**値は iOS を正本**とし、勝手に変えない。
 */
enum class AppTheme(val displayName: String, val hint: String) {
    Peach("ピーチ", "やわらかピーチ&コーラル"),         // デフォルト (女性向け)
    Sky("スカイ", "クールなブルー系"),                  // 男性向け
    Midnight("ミッドナイト", "目に優しい暗め配色"),       // ダーク
    Sunshine("サンシャイン", "明るく元気な黄色系"),       // 黄色系
    Forest("フォレスト", "落ち着きのある自然系");         // 自然系

    val background: Color
        get() = when (this) {
            Peach -> Color(1.00f, 0.97f, 0.93f)
            Sky -> Color(0.94f, 0.97f, 1.00f)
            Midnight -> Color(0.10f, 0.11f, 0.14f)
            Sunshine -> Color(1.00f, 0.98f, 0.90f)
            Forest -> Color(0.95f, 0.97f, 0.93f)
        }

    val surface: Color
        get() = when (this) {
            Peach -> Color(1.00f, 0.99f, 0.96f)
            Sky -> Color(1.00f, 1.00f, 1.00f)
            Midnight -> Color(0.17f, 0.18f, 0.22f)
            Sunshine -> Color(1.00f, 1.00f, 0.96f)
            Forest -> Color(1.00f, 1.00f, 0.98f)
        }

    val primary: Color
        get() = when (this) {
            Peach -> Color(1.00f, 0.62f, 0.55f)
            Sky -> Color(0.36f, 0.62f, 0.92f)
            Midnight -> Color(0.65f, 0.78f, 1.00f)
            Sunshine -> Color(1.00f, 0.78f, 0.30f)
            Forest -> Color(0.42f, 0.66f, 0.52f)
        }

    val primaryDeep: Color
        get() = when (this) {
            Peach -> Color(0.84f, 0.33f, 0.30f)
            Sky -> Color(0.18f, 0.40f, 0.78f)
            Midnight -> Color(0.45f, 0.62f, 0.95f)
            Sunshine -> Color(0.85f, 0.58f, 0.12f)
            Forest -> Color(0.22f, 0.50f, 0.32f)
        }

    val secondary: Color
        get() = when (this) {
            Peach -> Color(0.96f, 0.85f, 0.74f)
            Sky -> Color(0.78f, 0.88f, 0.98f)
            Midnight -> Color(0.30f, 0.34f, 0.42f)
            Sunshine -> Color(0.98f, 0.90f, 0.70f)
            Forest -> Color(0.82f, 0.92f, 0.84f)
        }

    val textPrimary: Color
        get() = when (this) {
            Peach -> Color(0.30f, 0.25f, 0.20f)
            Sky -> Color(0.16f, 0.20f, 0.30f)
            Midnight -> Color(0.95f, 0.95f, 0.97f)
            Sunshine -> Color(0.35f, 0.28f, 0.12f)
            Forest -> Color(0.18f, 0.26f, 0.20f)
        }

    val textSecondary: Color
        get() = when (this) {
            Peach -> Color(0.54f, 0.47f, 0.39f)
            Sky -> Color(0.40f, 0.46f, 0.55f)
            Midnight -> Color(0.70f, 0.72f, 0.78f)
            Sunshine -> Color(0.55f, 0.48f, 0.32f)
            Forest -> Color(0.42f, 0.50f, 0.42f)
        }

    val success: Color
        get() = when (this) {
            Peach -> Color(0.55f, 0.78f, 0.55f)
            Sky -> Color(0.36f, 0.78f, 0.66f)
            Midnight -> Color(0.40f, 0.85f, 0.65f)
            Sunshine -> Color(0.60f, 0.80f, 0.40f)
            Forest -> Color(0.45f, 0.74f, 0.48f)
        }

    val restDay: Color
        get() = when (this) {
            Peach -> Color(0.70f, 0.80f, 0.95f)
            Sky -> Color(0.60f, 0.78f, 0.95f)
            Midnight -> Color(0.45f, 0.55f, 0.78f)
            Sunshine -> Color(0.78f, 0.84f, 0.95f)
            Forest -> Color(0.70f, 0.85f, 0.90f)
        }

    val missed: Color
        get() = when (this) {
            Peach -> Color(0.86f, 0.47f, 0.45f)
            Sky -> Color(0.85f, 0.55f, 0.55f)
            Midnight -> Color(0.95f, 0.45f, 0.55f)
            Sunshine -> Color(0.88f, 0.50f, 0.30f)
            Forest -> Color(0.78f, 0.50f, 0.45f)
        }

    val historyAccent: Color
        get() = when (this) {
            Peach -> Color(0.62f, 0.49f, 0.86f)
            Sky -> Color(0.45f, 0.58f, 0.85f)
            Midnight -> Color(0.75f, 0.60f, 0.92f)
            Sunshine -> Color(0.85f, 0.55f, 0.45f)
            Forest -> Color(0.55f, 0.65f, 0.45f)
        }

    val settingsAccent: Color
        get() = when (this) {
            Peach -> Color(0.36f, 0.62f, 0.72f)
            Sky -> Color(0.30f, 0.55f, 0.78f)
            Midnight -> Color(0.55f, 0.70f, 0.85f)
            Sunshine -> Color(0.65f, 0.55f, 0.40f)
            Forest -> Color(0.45f, 0.62f, 0.55f)
        }

    val chipBackground: Color
        get() = when (this) {
            Peach -> Color(1.00f, 0.91f, 0.86f)
            Sky -> Color(0.88f, 0.94f, 1.00f)
            Midnight -> Color(0.22f, 0.25f, 0.32f)
            Sunshine -> Color(1.00f, 0.94f, 0.78f)
            Forest -> Color(0.92f, 0.96f, 0.88f)
        }

    fun categoryColor(category: WorkoutCategory): Color = when (category) {
        WorkoutCategory.Cardio -> Color(0.36f, 0.62f, 0.72f)
        WorkoutCategory.Strength -> primaryDeep
        WorkoutCategory.Yoga -> Color(0.58f, 0.55f, 0.82f)
        WorkoutCategory.Stretch -> Color(0.42f, 0.66f, 0.52f)
        WorkoutCategory.FasciaRelease -> Color(0.85f, 0.55f, 0.40f)
        WorkoutCategory.Other -> historyAccent
    }

    /** midnight のみダーク扱い(iOS preferredColorScheme と一致)。 */
    val isDark: Boolean get() = this == Midnight
}
